### ISSUE TITLE
tied variables don't work with .= <>

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#8346]
Migrated from [rt.perl.org#38631](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=38631) (status was 'resolved')

Searchable as RT38631$

**Comment by p5pRT** [comment]
### From @nwc10

#### Created by @nwc10



$ cat tie\_rcatline\.pl
\#\!/usr/bin/perl \-w
use strict;
use vars qw\($a $b $v $x\);

require Tie&#8203;::Scalar;

tie $a\, 'Tie&#8203;::StdScalar'\, "A&#8203;: ";
tie $b\, 'Tie&#8203;::StdScalar'\, "B&#8203;: ";

tie $v\, 'Tie&#8203;::StdScalar'\, "5&#8203;: ";
tie $x\, 'Tie&#8203;::StdScalar'\, "10&#8203;: ";

my $junk = $a;
my $junk2 = $v;

$a \.= \<DATA>;
$b \.= \<DATA>;
$v \.= \<DATA>;
$x \.= \<DATA>;

print $a;
print $b;
print $v;
print $x;

\_\_DATA\_\_
One
Two
Three
Four
$ perl tie\_rcatline\.pl
A&#8203;: One
Two
5&#8203;: Three
Four


This would be a bug :\-\)

Worked with 5\.6\.1\. I suspect that it's because 5\.8\.0 was optimised to use
the rcatline op\. I also think I know why it's happening \- a check for SvOK\(\)
in do\_readline\, without any preceding magic get\. Although in turn I don't
think that the solution is necessarily quite as simple as adding a magic get\.

Nicholas Clark

<details><summary>Perl Info</summary>

```

Flags:
    category=core
    severity=low

Site configuration information for perl v5.8.0:

Configured by nwc10 at Fri Jan 13 14:56:47 GMT 2006.

Summary of my perl5 (revision 5.0 version 8 subversion 0) configuration:
  Platform:
    osname=freebsd, osvers=5.4-stable, archname=i386-freebsd
    uname='freebsd colon.colondot.net 5.4-stable freebsd 5.4-stable #3: sat oct 15 10:50:35 bst 2005 mbm@colonii.colondot.net:usrobjusrsrcsyscolon i386 '
    config_args='-Dprefix=~/Reference/5.8.0 -Doptimize=-Os -Uuse64bitint -Uusethreads -Dcc=ccache gcc -Dld=gcc -Ubincompat5005 -Uinstallusrbinperl -Dcf_email=nick@ccl4.org -Dperladmin=nick@ccl4.org -Dinc_version_list=  -Dinc_version_list_init=0 -Uinstallman1dir -Uinstallman3dir -Dmalloctype=void * -Dfreetype=void -de'
    hint=recommended, useposix=true, d_sigaction=define
    usethreads=undef use5005threads=undef useithreads=undef usemultiplicity=undef
    useperlio=define d_sfio=undef uselargefiles=define usesocks=undef
    use64bitint=undef use64bitall=undef uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='ccache gcc', ccflags ='-DHAS_FPSETMASK -DHAS_FLOATINGPOINT_H -fno-strict-aliasing -I/usr/local/include',
    optimize='-Os',
    cppflags='-DHAS_FPSETMASK -DHAS_FLOATINGPOINT_H -fno-strict-aliasing -I/usr/local/include'
    ccversion='', gccversion='3.4.2 [FreeBSD] 20040728', gccosandvers=''
    intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=1234
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=4, prototype=define
  Linker and Libraries:
    ld='gcc', ldflags ='-Wl,-E  -L/usr/local/lib'
    libpth=/usr/lib /usr/local/lib
    libs=-lgdbm -lm -lc -lcrypt -lutil
    perllibs=-lm -lc -lcrypt -lutil
    libc=, so=so, useshrplib=false, libperl=libperl.a
    gnulibc_version=''
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags=' '
    cccdlflags='-DPIC -fPIC', lddlflags='-shared  -L/usr/local/lib'

Locally applied patches:
    


@INC for perl v5.8.0:
    /export/home/nwc10/Reference/5.8.0/lib/perl5/5.8.0/i386-freebsd
    /export/home/nwc10/Reference/5.8.0/lib/perl5/5.8.0
    /export/home/nwc10/Reference/5.8.0/lib/perl5/site_perl/5.8.0/i386-freebsd
    /export/home/nwc10/Reference/5.8.0/lib/perl5/site_perl/5.8.0
    /export/home/nwc10/Reference/5.8.0/lib/perl5/site_perl
    .


Environment for perl v5.8.0:
    HOME=/export/home/nwc10
    LANG (unset)
    LANGUAGE (unset)
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/export/home/nwc10/bin:/sbin:/bin:/usr/sbin:/usr/bin:/usr/games:/usr/local/sbin:/usr/local/bin:/usr/X11R6/bin:/export/home/nwc10/bin:/usr/local/sbin:/sbin:/usr/sbin
    PERL_BADLANG (unset)
    SHELL=/usr/local/bin/bash

```
</details>

**Comment by p5pRT** [comment]
### From @andk

>>>>> On Sat\, 25 Feb 2006 03&#8203;:40&#8203;:17 \-0800\, Nicholas Clark \(via RT\) \<perlbug\-followup@&#8203;perl\.org> said&#8203;:

&nbsp;  > Worked with 5\.6\.1\. I suspect that it's because 5\.8\.0 was optimised to use
&nbsp;  > the rcatline op\. I also think I know why it's happening \- a check for SvOK\(\)
&nbsp;  > in do\_readline\, without any preceding magic get\. Although in turn I don't
&nbsp;  > think that the solution is necessarily quite as simple as adding a magic get\.

Just to confirm what you supect&#8203;:

\-\-\-\-Output of \.\.\./pGDSpKt/perl\-5\.7\.2@&#8203;11633/bin/perl\-\-\-\-
A&#8203;: One
B&#8203;: Two
5&#8203;: Three
10&#8203;: Four

\-\-\-\-EOF \($?='0'\)\-\-\-\-
\-\-\-\-Output of \.\.\./pNiSb5r/perl\-5\.7\.2@&#8203;11634/bin/perl\-\-\-\-
A&#8203;: One
Two
5&#8203;: Three
Four

\-\-\-\-EOF \($?='0'\)\-\-\-\-

Change 11634 by jhi@&#8203;alpha on 2001/08/11 14&#8203;:59&#8203;:22

&nbsp;    Subject&#8203;: \[PATCH\] turn $a\.=\<F> into an RCATLINE
&nbsp;    From&#8203;: Abhijit Menon\-Sen \<ams@&#8203;wiw\.org>
&nbsp;    Date&#8203;: Sat\, 11 Aug 2001 16&#8203;:31&#8203;:04 \+0530
&nbsp;    Message\-ID&#8203;: \<20010811163104\.Q526@&#8203;lustre\.dyn\.wiw\.org>




\-\- 
andreas

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From @rgs

Solved by change \#29361 in bleadperl\.

**Comment by p5pRT** [comment]
### From @rgarcia

Solved by change \#29361 in bleadperl\.

**Comment by p5pRT** [comment]
@rgs - Status changed from 'open' to 'resolved'
